libnavui-androidauto/src/main/java/com/mapbox/androidauto/map/MapboxCarMapLoader.kt

```diff
@@ -1,19 +1,32 @@
-package com.mapbox.navigation.qa_test_app.car
+package com.mapbox.androidauto.map
 
+import androidx.car.app.CarContext
+import androidx.car.app.Session
 import com.mapbox.androidauto.internal.logAndroidAuto
 import com.mapbox.androidauto.internal.logAndroidAutoFailure
 import com.mapbox.maps.MapboxExperimental
 import com.mapbox.maps.MapboxMap
+import com.mapbox.maps.extension.androidauto.MapboxCarMap
 import com.mapbox.maps.extension.androidauto.MapboxCarMapObserver
 import com.mapbox.maps.extension.androidauto.MapboxCarMapSurface
 import com.mapbox.maps.extension.observable.eventdata.MapLoadingErrorEventData
 import com.mapbox.maps.plugin.delegates.listeners.OnMapLoadErrorListener
 import com.mapbox.navigation.ui.maps.NavigationStyles
 
+/**
+ * Default map style loader that is designed for navigation. This map loader assumes you want to
+ * keep map styles in sync with the [CarContext.isDarkMode].
+ *
+ * To use, create an instance and register it to [MapboxCarMap.registerObserver]. It will
+ * automatically load the map for as long as it is registered. To disable you can unregister it
+ * with [MapboxCarMap.unregisterObserver].
+ */
 @OptIn(MapboxExperimental::class)
-class MainCarMapLoader : MapboxCarMapObserver {
+class MapboxCarMapLoader : MapboxCarMapObserver {
 
     private var mapboxMap: MapboxMap? = null
+    private var lightStyleOverride: String? = null
```

Comment:
Can we add support for [`StyleExtension`](https://github.com/mapbox/mapbox-maps-android/blob/1243eec9e3bdb1d46f2b6139937375a5c9b107da/sdk/src/main/java/com/mapbox/maps/MapboxMap.kt#L307)? It gives more options when loading a style.

Reply:
Yep that adds support for a lot of things! added